db/sync.go

```diff
@@ -0,0 +1,236 @@
+package db
+
+import (
+	"database/sql"
+	"strings"
+
+	log "github.com/sirupsen/logrus"
+)
+
+const lastModifiedField = "last_modified"
+
+func Sync(sourceDB, targetDB *sql.DB) error {
+	tables, err := getTablesWithLastModifiedFrom(sourceDB)
+	if err != nil {
+		log.WithError(err).Error("Error in getting list of tables with last_modified field from the source DB")
+		return err
+	}
+
+	for _, table := range tables {
+		err := syncTable(table, sourceDB, targetDB)
+		if err != nil {
+			log.WithError(err).WithField("table", table).Error("Error in synchronising table")
+			return err
```

Comment:
Just a thought;

We immediately give up synching the whole list of tables when we fail a specific one. If we continue synching (trying at least) the rest of the tables, will there be any kind of benefit for us? Or any unwanted side effects?

Reply:
I don't see any unwanted side effect. Since Aurora is our "golden store" for PAC, I was thinking that is better to fail immediately, then fix manually a potential data/sxhema inconsistency that blocks the synchronisation.

Reply:
👍